### PR TITLE
refactor: move invalid-block tests back to main testing function

### PR DESCRIPTION
**Pull request by StephenButtolph**
`TestBlockGetters` is confusing, as it tests _almost_ all of the cases for the block getters. It makes more sense to me to have that test all of the cases of the function, rather than splitting out some (not so special) edge cases to a different tester

### CODE REVIEW

File: sae/rpc_test.go [hunk 1]
```diff
@@ -257,11 +257,17 @@ func TestTxPoolNamespace(t *testing.T) {
...
 
 	ctx, sut := newSUT(t, 1, opt)
-	genesis := sut.lastAcceptedBlock(t)
+
+	t.Run("unknown_hashes", func(t *testing.T) {
+		sut.testGetByUnknownHash(ctx, t)
+	})
+	t.Run("unknown_numbers", func(t *testing.T) {
+		sut.testGetByUnknownNumber(ctx, t)
+	})
```
Comment (ARR4N) [inline sae/rpc_test.go]: I'm much happier with these being up here and immediately after `newSUT()` so they clearly signal "you don't need to remember anything".
